docs/website/postprocess_dialect_docs.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# MLIR dialect markdown document postprocessor for website generation.
+#
+# Usage (typically invoked by generate_extra_files.sh):
+#   postprocess_dialect_docs.py ${WEBSITE_DOCS_BUILD_DIRECTORY}
+
+import argparse
+import os
+import fileinput
+import re
+
+
+def main(args):
+    directory = args.directory
+    files = [os.path.join(directory, f) for f in os.listdir(directory)]
```

Comment:
I think the more modern way to do this would be `pathlib.Path.iterdir`


https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir

```suggestion
    files = list(pathlib.Path(directory).iterdir())
```

If `fileinput.input` will accept any iterable, you could skip the list conversion and maybe even inline it too. Although you may actually want to filter for things actually being a file. All of these methods also return directories (I know you don't expect there to be any in this case)

Reply:
> If `fileinput.input` will accept any iterable, you could skip the list conversion and maybe even inline it too.

I'll keep the list of files in a separate variable rather than inline, in case I want to add another pass through the files. For example, if I wanted to add frontmatter like
```markdown
---
hide:
  - tags
tags:
  - MLIR
---

<!-- Autogenerated by mlir-tblgen; don't manually edit -->
# 'hal' Dialect
```

I think I would add another loop over the files, rather than put more code in 
```python
    with fileinput.input(files=files, inplace=True) as f:
        for line in f:
            # (edit `line` here)
            print(line, end="")
```